tests/scripts/thread-cert/Cert_5_7_03_CoapDiagCommands.py

```diff
@@ -0,0 +1,266 @@
+        else:
+            ROUTER = pv.vars['ROUTER']
+            pv.verify_attached('ROUTER', 'LEADER')
+            pv.verify_attached('SED', 'ROUTER', 'MTD')
+            pv.verify_attached('MED', 'ROUTER', 'MTD')
+            pv.verify_attached('DUT', 'ROUTER', 'FTD-ED')
+
+        # Step 2: Leader sends DIAG_GET.qry to the Realm-Local All-Thread-Nodes
+        #         multicast address containing the requested diagnostic TLVs:
+        #         CoAP Response Code
+        #             2.04 Changed
+        #         CoAP Payload
+        #             TLV Type 0 - MAC Extended Address (64- bit)
+        #             TLV Type 1 - MAC Address (16-bit)
+        #             TLV Type 2 - Mode (Capability information)
+        #             TLV Type 6 – Leader Data
+        #             TLV Type 7 – Network Data
+        #             TLV Type 8 – IPv6 address list
+        #             TLV Type 17 – Channel Pagesi
+        #
+        #         if DUT is Router, contianing the following as well:
+        #             TLV Type 4 – Connectivity
+        #             TLV Type 5 – Route64
+        #             TLV Type 16 – Child Table
+        _qr_pkt = pkts.filter_wpan_src64(LEADER).\
+            filter_ipv6_dst(REALM_LOCAL_All_THREAD_NODES_MULTICAST_ADDRESS).\
+            filter_coap_request(DIAG_GET_QRY_URI).\
+            must_next()
+        dut_payload_tlvs = {
+            DG_TYPE_LIST_TLV, DG_MAC_EXTENDED_ADDRESS_TLV, DG_MAC_ADDRESS_TLV, DG_MODE_TLV, DG_LEADER_DATA_TLV,
+            DG_NETWORK_DATA_TLV, DG_IPV6_ADDRESS_LIST_TLV, DG_CHANNEL_PAGES_TLV
+        }
+        if self.TOPOLOGY[ROUTER1]['name'] == 'DUT':
+            dut_payload_tlvs.update({DG_CONNECTIVITY_TLV, DG_ROUTE64_TLV, DG_CHILD_TABLE_TLV})
+            _qr_pkt.must_verify(lambda p: dut_payload_tlvs == set(p.thread_diagnostic.tlv.type))
+
+            # Step 3: The DUT automatically responds with a DIAG_GET.ans response
+            #         MUST contain the requested diagnostic TLVs:
+            #             TLV Type 0 - MAC Extended Address (64- bit)
+            #             TLV Type 1 - MAC Address (16-bit)
+            #             TLV Type 2 - Mode (Capability information)
+            #             TLV Type 4 – Connectivity
+            #             TLV Type 5 – Route64
+            #             TLV Type 6 – Leader Data
+            #             TLV Type 7 – Network Data
+            #             TLV Type 8 – IPv6 address list
+            #             TLV Type 16 – Child Table
+            #             TLV Type 17 – Channel Pages
+            dut_payload_tlvs.remove(DG_TYPE_LIST_TLV)
+            pkts.filter_wpan_src64(DUT).\
+                filter_ipv6_dst(LEADER_RLOC).\
+                filter_coap_request(DIAG_GET_ANS_URI).\
+                filter(lambda p:
+                       dut_payload_tlvs == set(p.thread_diagnostic.tlv.type) and\
+                       {str(p.wpan.src64), colon_hex(dut_addr16, 2), '0f'}
```

Comment:
```suggestion
                       {str(p.wpan.src64), rloc16_to_hex(DUT_RLOC16), '0f'}
```

nit : Why not convert RLOC16 to hexstr directly?

Reply:
this common function is used to convert hex str to colon separated str with an interval, it might not be only for RLOC16 but also for any hex str with user defined interval